### PR TITLE
add integration tests for NAMESPACE_WAITING_STARTED and NAMESPACE_WATCHING_STOPPED events

### PR DESCRIPTION
**Pull request by xiancao**
Write integration test cases for the function added in owls-86455.
Two events are added: NAMESPACE_WAITING_STARTED and NAMESPACE_WATCHING_STOPPED.
1. When the operator starts, one NAMESPACE_WAITING_STARTED event should be generated for each namespace the operator manages.
2. When the operator stops to manage a namespace, one NAMESPACE_WATCHING_STOPPED event should be generated.
Tested all domainNamespaceSelectionStrategy types that the operator supports

Jenkins result: 
https://build.weblogick8s.org:8443/view/all/job/weblogic-kubernetes-operator-kind-new/4778/
https://build.weblogick8s.org:8443/view/all/job/weblogic-kubernetes-operator-kind-new/4777/testReport/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java [hunk 1]
```diff
@@ -515,76 +512,242 @@ public void testDomainK8SEventsDelete() {
...
+    domainNamespaces.add(domainNamespace2);
+    opParams = opParams.domainNamespaces(domainNamespaces).enableClusterRoleBinding(enableClusterRoleBinding);
+    upgradeAndVerifyOperator(opNamespace, opParams);
+
     logger.info("verify NamespaceWatchingStarted event is logged");
     checkEvent(opNamespace, domainNamespace2, null, NAMESPACE_WATCHING_STARTED, "Normal", timestamp);
+
+    timestamp = now();
+
+    logger.info("Removing domain namespace in the operator watch list");
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java:552]: Please indicate which domain space is removed from watch

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java [hunk 2]
```diff
@@ -515,76 +512,242 @@ public void testDomainK8SEventsDelete() {
...
-  public void testK8SEventsStopWatchingNS() {
+  @Order(13)
+  @ParameterizedTest
+  @ValueSource(booleans = { true, false })
+  public void testK8SEventsStartStopWatchingNSWithLabelSelector(boolean enableClusterRoleBinding) {
+    logger.info("testing testK8SEventsStartStopWatchingNSWithLabelSelector with enableClusterRoleBinding={0}",
+        enableClusterRoleBinding);
     OffsetDateTime timestamp = now();
+
+    logger.info("Adding a new domain namespace in the operator watch list");
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java:589]:
```suggestion
    logger.info("Adding a new domain namespace {0} in the operator watch list", domainNamespace3);
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java [hunk 3]
```diff
@@ -515,76 +512,242 @@ public void testDomainK8SEventsDelete() {
...
+    checkEvent(opNamespace, domainNamespace3, null, NAMESPACE_WATCHING_STARTED, "Normal", timestamp);
+
+    // verify there is no event logged in domainNamespace4
+    logger.info("verify NamespaceWatchingStarted event is not logged in " + domainNamespace4);
+    assertFalse(domainEventExists(opNamespace, domainNamespace4, null, NAMESPACE_WATCHING_STARTED,
+        "Normal", timestamp), "domain event " + NAMESPACE_WATCHING_STARTED + " is logged in "
+        + domainNamespace4 + ", expected no such event will be logged");
+
+    timestamp = now();
+    logger.info("Removing domain namespace in the operator watch list");
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java:613]:
```suggestion
    logger.info("Removing domain namespace {0} in the operator watch list", domainNamespace3);
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java [hunk 4]
```diff
@@ -515,76 +512,242 @@ public void testDomainK8SEventsDelete() {
...
+  }
+
+  /**
+   * Test verifies the operator logs a NamespaceWatchingStarted event in the respective domain namespace
+   * when it starts watching a new domain namespace with domainNamespaceSelectionStrategy set to RegExp and
+   * operator logs a NamespaceWatchingStopped event in the respective domain namespace
+   * when it stops watching a domain namespace.
+   * The test upgrades the operator instance through helm to add or remove another domain namespace
+   * in the operator watch list.
+   *<p>
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java:633]: can you please document how the operator uses RegExp(the same for Dedicated as well)  to start/stop watching namespace? A example would be nice.

**Review by sankarpn — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java (hunk 1 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java (hunk 2 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java (hunk 3 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java (hunk 4 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java:638]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java [hunk 5]
```diff
@@ -45,27 +47,97 @@
...
+              NAMESPACE_WATCHING_STOPPED, domainNamespace, STOP_MANAGING_NAMESPACE, opNamespace);
+          return true;
+        } else {
+          logger.info("Did not get the {0} event in namespace {1} or {2} event in namespace {3}",
+              NAMESPACE_WATCHING_STOPPED, domainNamespace, STOP_MANAGING_NAMESPACE, opNamespace);
+
+          // check if there is a warning message in operator's log
+          String operatorPodName = getOperatorPodName(OPERATOR_RELEASE_NAME, opNamespace);
+          String expectedErrorMsg = String.format(
+              "Cannot create NamespaceWatchingStopped event in namespace %s due to an authorization error",
```
[the remark this answers is not in the document]
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java]: fixed

**Review by doxiao — APPROVED** [review]

**Review by sankarpn — APPROVED** [review]
LGTM

**Pull request merged**